test/helpers/kubectl.go

```diff
@@ -418,6 +418,14 @@ func (kub *Kubectl) NamespaceDelete(name string) *CmdRes {
 // the aforementioned desired state within timeout seconds. Returns false and
 // an error if the command failed or the timeout was exceeded.
 func (kub *Kubectl) WaitforPods(namespace string, filter string, timeout time.Duration) error {
+
+	data, err := kub.GetPods(namespace, filter).Filter("{.items[*].metadata.deletionTimestamp}")
+	if err != nil {
+		return fmt.Errorf("Cannot get pods with filter '%s': %s", filter, err)
+	}
+	if data.String() != "" {
+		return fmt.Errorf("There is some pods with filter %s that are mark to delete", filter)
+	}
 	body := func() bool {
 		var jsonPath = "{.items[*].status.containerStatuses[*].ready}"
```

Comment:
Instead of returning an error, why not just only take pods that do not have a deletion timestamp?